lib/experimental/fonts-api/class-wp-fonts-theme-json-handler.php

```diff
@@ -1,52 +1,159 @@
 <?php
 /**
- * Bootstraps Global Styles.
+ * Webfonts API: Provider for locally-hosted fonts.
  *
- * @package gutenberg
+ * @package    WordPress
+ * @subpackage Fonts API
+ * @since      X.X.X
  */
 
-if ( ! function_exists( 'gutenberg_register_fonts_from_theme_json' ) ) {
+if ( class_exists( 'WP_Fonts_Theme_Json_Handler' ) ) {
+	return;
+}
+
+/**
+ * Wrapper class that contains several static methods
+ * to handle fonts defined in theme.json.
+ *
+ * @since X.X.X
+ */
+class WP_Fonts_Theme_Json_Handler {
```

Comment:
I'm rethinking this new class. Since I first wrote the issue's scope, a new resolver class was merged into the API. I'm thinking the static functions here can be merged into that resolver class instead. Why? These methods are doing resolver work. Rather than adding another class, I propose consolidating the functionality into the resolver.

To retain @anton-vlasenko's work, I'll open a new PR but cherry-pick Anton's work as a starting point.